sql/core/src/test/scala/org/apache/spark/sql/DataFrameNaFunctionsSuite.scala

```diff
@@ -194,4 +194,45 @@ class DataFrameNaFunctionsSuite extends QueryTest with SharedSQLContext {
     assert(out1(4) === Row("Amy", null, null))
     assert(out1(5) === Row(null, null, null))
   }
+
+  test("Spark-12231: dropna with partitionBy and groupBy") {
+    withTempPath { dir =>
+      val df = sqlContext.range(10)
+      val df1 = df.withColumn("a", $"id".cast("int"))
+      df1.write.partitionBy("id").parquet(dir.getCanonicalPath)
+      val df2 = sqlContext.read.parquet(dir.getCanonicalPath)
+      val group = df2.na.drop().groupBy().count().collect()
```

Comment:
This file is not a great example of this, but we should always use `checkAnswer` in unit tests because it provides much better failure messages than the default ScalaTest asserts.